### ISSUE TITLE
bug: terraform init force-copy prompts when using workspaces

### ISSUE THREAD

**Issue by dmccaffery** [hashicorp/terraform#29239]
When using the `-force-copy` option with `terraform init -migrate-state` within a root module containing workspaces, a
prompt that requires acknowledgement that all workspaces will be migrated is required. There is currently no way to
automate the migration of state when using workspaces, such as in a build pipeline.

Simple reproduction: https://github.com/dmccaffery/terraform-force-migrate-workspace

### Terraform Version

```
Terraform v1.0.3
on darwin_arm64
+ provider registry.terraform.io/hashicorp/null v3.1.0
```

### Terraform Configuration Files

```terraform
resource "null_resource" "example" {}

terraform {
  required_version = "~> 1.0"

  backend "local" {
  }

  required_providers {
    null = {
      source  = "hashicorp/null"
      version = "~> 3.0"
    }
  }
}
```

backends/local.hcl (for this example):

```hcl
path = "something/different.tfstate"
```

### Debug Output

https://gist.github.com/dmccaffery/38967d377c89624157691926fd56f27f

### Crash Output

N/A

### Expected Behavior

1. state is migrated successfully without requiring a prompt

### Actual Behavior

1. state does not migrate and an error occurs because a prompt is required.

### Steps to Reproduce

Simple reproduction: https://github.com/dmccaffery/terraform-force-migrate-workspace

1. initialise and apply some state in a workspace

    ```hcl
    terraform init
    terraform workspace new example
    terraform apply -auto-approve -input false
    ```

2. attempt to migrate state using a new backend config

    ```hcl
    terraform init -migrate-state -force-copy -input=false -backend-config=backends/local.tf
    ```

### Additional Context

Force copy requires interactive session whenever workspaces are used.

### References

- #17663 - issue was closed but I am able to consistently reproduce

**Comment by alisdair** [comment]
Hi @dmccaffery, thank you for this thorough bug report and reproduction case. I've reproduced the issue locally and it looks to me like this behaviour might just be an oversight. There's a proposed fix under review at #29438.

**Comment by alisdair** [comment]
The fix for this should go out in the next 1.0 patch release. Thanks again!

**Comment by dmccaffery** [comment]
No worries! Thanks for the prompt fix. Have a great weekend!
